src/context.ts

```diff
@@ -32,6 +38,164 @@ export type AliasProps<U> = {
+            const txt = msg.text ?? msg.caption;
+            return match(ctx, txt, trg);
+        };
+    },
+    command(command) {
+        const hasEntities = checker.filterQuery(":entities:bot_command");
+        const atCommands = new Set<string>();
+        const noAtCommands = new Set<string>();
+        toArray(command).forEach((cmd) => {
+            if (cmd.startsWith("/")) {
+                throw new Error(
+                    `Do not include '/' when registering command handlers (use '${
+                        cmd.substring(1)
+                    }' not '${cmd}')`,
+                );
+            }
+            const set = cmd.indexOf("@") === -1 ? noAtCommands : atCommands;
+            set.add(cmd);
+        });
+        return <C extends Context>(ctx: C): ctx is CommandContext<C> => {
+            if (!hasEntities(ctx)) return false;
+            const msg = ctx.message ?? ctx.channelPost;
+            const txt = msg.text ?? msg.caption;
+            return msg.entities.some((e) => {
+                if (e.type !== "bot_command") return false;
+                if (e.offset !== 0) return false;
+                const cmd = txt.substring(1, e.length);
+                if (noAtCommands.has(cmd) || atCommands.has(cmd)) {
+                    ctx.match = txt.substring(cmd.length + 1).trimStart();
+                    return true;
+                }
+                const index = cmd.indexOf("@");
+                if (index === -1) return false;
+                const atTarget = cmd.substring(index + 1);
+                if (atTarget !== ctx.me.username) return false;
+                const atCommand = cmd.substring(0, index);
+                if (noAtCommands.has(atCommand)) {
+                    ctx.match = txt.substring(cmd.length + 1).trimStart();
+                    return true;
+                }
+                return false;
+            });
+        };
+    },
+    chatType<T extends Chat["type"]>(chatType: MaybeArray<T>) {
+        const set = new Set<Chat["type"]>(toArray(chatType));
+        return <C extends Context>(ctx: C): ctx is ChatTypeContext<C, T> =>
+            ctx.chat?.type !== undefined && set.has(ctx.chat.type);
+    },
+    callbackQuery(trigger) {
+        const hasCallbackQuery = checker.filterQuery("callback_query:data");
+        const trg = triggerFn(trigger);
+        return <C extends Context>(ctx: C): ctx is CallbackQueryContext<C> =>
+            hasCallbackQuery(ctx) && match(ctx, ctx.callbackQuery.data, trg);
+    },
+    gameQuery(trigger) {
+        const hasCallbackQuery = checker.filterQuery(
+            "callback_query:game_short_name",
+        );
+        const trg = triggerFn(trigger);
+        return <C extends Context>(ctx: C): ctx is GameQueryContext<C> =>
+            hasCallbackQuery(ctx) &&
+            match(ctx, ctx.callbackQuery.game_short_name, trg);
+    },
+    inlineQuery(trigger) {
+        const hasCallbackQuery = checker.filterQuery("inline_query");
```

Comment:
Maybe hasInlineQuery?

Reply:
No. That would lead to `Context.has.hasInlineQuery`, which is redundant.

Reply:
No, i mean the const

Reply:
Oh. Good point, I'll fix it, thanks..